crates/python/qcs_sdk/compiler/quilc.pyi

```diff
@@ -233,6 +211,55 @@ async def compile_program_async(
     """
     ...
 
+@final
+class CompilationResult:
+    @property
+    def program(self) -> str:
+        """ """
+        ...
+    @property
+    def native_quil_metadata(self) -> Optional[NativeQuilMetadata]:
+        """ """
+        ...
+
+class NativeQuilMetadata:
```

Comment:
Not sure if this is needed - if the type checker works then /shrug

```suggestion
@final
class NativeQuilMetadata:
```

Reply:
Ah yeah, it is. Helpful hint to have since the error you get when trying to use it as a base class is a bit obtuse.